django/db/models/functions/text.py

```diff
@@ -2,6 +2,22 @@
 from django.db.models.functions import Coalesce
 
 
+class BytesToCharFieldConversionMixin:
+    """
+    Add conversion to char fields. Some databases engines such as MySQL returns
```

Comment:
My suggestion is:
```
Convert CharField results from bytes to str.

MySQL returns long data types (bytes) instead of chars when it can't
determine the length of the result string. For example:
    LPAD(column1, CHAR_LENGTH(column2), ' ')
 returns the LONGTEXT (bytes) instead of VARCHAR.
```
I'm not sure if it would make a meaningful difference, but a feature flag might be useful since `isinstance` can be somewhat expensive, I think.